File: tests/pywin/test_pywin32.py

import win32com_.client.gencache as gencache
import pandas
from pathlib import Path
import pandas.core.groupby.generic #Added for inference

# Read in the remote data file
df: pandas.DataFrame = pandas.read_csv("https://github.com/chris1610/pbpython/blob/master/data/sample-sales-tax.csv?raw=True")

# Define the full path for the output file
out_file = Path.cwd() / "tests/pywin/tax_summary.xlsx"
# Do some summary calcs
# In the real world, this would likely be much more involved
df_summary: pandas.DataFrame = df.groupby('category')['ext price', 'Tax amount'].sum()

# Save the file as Excel
df_summary.to_excel(out_file)

# Open up Excel and make it visible
excel = gencache.EnsureDispatch('Excel.Application')
excel.Visible = True

# Open up the file
excel.Workbooks.Open(out_file)

# Wait before closing it
_ = input("Press enter to close Excel")
excel.Application.Quit()

File: tests/pywin-py2many/test_pywin32.jl
using Pandas
include("win32com_/client/gencache.jl")

using pathlib: Path

df::pandas.DataFrame = read_csv(
    pandas,
    "https://github.com/chris1610/pbpython/blob/master/data/sample-sales-tax.csv?raw=True",
)
out_file = pwd() / "tests/pywin/tax_summary.xlsx"
df_summary::pandas.DataFrame = sum(groupby("category")[("ext price", "Tax amount")+1])
to_excel(out_file)
excel = EnsureDispatch(gencache, "Excel.Application")
excel.Visible = true
Open(excel.Workbooks, out_file)
_ = input("Press enter to close Excel")
Quit(excel.Application)
